diff --git a/headers/linux/if_xdp.h b/headers/linux/if_xdp.h
index c6569e9..0214cc2 100644
--- a/headers/linux/if_xdp.h
+++ b/headers/linux/if_xdp.h
@@ -25,6 +25,7 @@
  * application.
  */
 #define XDP_USE_NEED_WAKEUP (1 << 3)
+#define XDP_USE_SG     (1 << 4)
 
 /* Flags for xsk_umem_config flags */
 #define XDP_UMEM_UNALIGNED_CHUNK_FLAG (1 << 0)
diff --git a/lib/libxdp/xsk.c b/lib/libxdp/xsk.c
index 566ceda..d0c46f8 100644
--- a/lib/libxdp/xsk.c
+++ b/lib/libxdp/xsk.c
@@ -876,6 +876,7 @@ static int __xsk_setup_xdp_prog(struct xsk_socket *xsk, int *xsks_map_fd)
 		if (err)
 			goto err_prog_load;
 
+		xdp_program__set_xdp_frags_support(ctx->xdp_prog, true);
 		err = xdp_program__attach(ctx->xdp_prog, ctx->ifindex,
 					  xsk_convert_xdp_flags(xsk->config.xdp_flags), 0);
 		if (err)
